lib/optimizely.rb

```diff
@@ -96,11 +96,15 @@ def activate(experiment_key, user_id, attributes = nil)
 
       return nil unless Optimizely::Helpers::Validator.inputs_valid?(
         {
-          experiment_key: experiment_key,
-          user_id: user_id
+          experiment_key: experiment_key
         }, @logger, Logger::ERROR
       )
 
+      unless user_id.is_a?(String)
```

Comment:
I'd rather we refactor a bit and avoid using this same logic everywhere in the code. Let's mimic what the C# SDK does here: https://github.com/optimizely/csharp-sdk/blob/master/OptimizelySDK/Optimizely.cs#L642

Reply:
Done.